test/integration/scheduler_perf/util.go

```diff
@@ -282,3 +323,51 @@ func (tc *throughputCollector) collect() []DataItem {
 
 	return []DataItem{throughputSummary}
 }
+
+// getExtensionPointDurationHist return a labeld Histogram for `scheduler_framework_extension_point_duration_seconds` metric.
+// TODO: Consolidate this method with `testutil.GetHistogramFromGatherer` to remove the duplicated code.
+func getExtensionPointDurationHist(gatherer metrics.Gatherer, metricName string) ([]labeledHist, error) {
+	var metricFamily *dto.MetricFamily
+	var hists []labeledHist
+
+	m, err := gatherer.Gather()
+	if err != nil {
+		return nil, err
+	}
+	for _, mFamily := range m {
+		if mFamily.GetName() == metricName {
+			metricFamily = mFamily
+			break
+		}
+	}
+
+	if metricFamily == nil {
+		return nil, fmt.Errorf("metric %q not found", metricName)
+	}
+
+	if metricFamily.GetMetric() == nil {
+		return nil, fmt.Errorf("metric %q is empty", metricName)
+	}
+
+	if len(metricFamily.GetMetric()) == 0 {
```

Comment:
this condition also checks for the previous if condition, we can remove the `== nil` check

Reply:
Done